packages/frontend/core/src/components/providers/workspace-side-effects.tsx

```diff
@@ -129,6 +137,20 @@ export const WorkspaceSideEffects = () => {
     };
   }, [globalDialogService]);
 
+  const graphqlService = useService(GraphQLService);
+  const eventSourceService = useService(EventSourceService);
+  const fetchService = useService(FetchService);
+
+  useEffect(() => {
+    setupAIProvider(
```

Comment:
seems `setupAIProvider` has some subscriptions that were not unsubscribed which were ok before. We may need to clean them up if having configurable servers.

Reply:
It seems that each time setupAIProvider() will overwrite the previous one?